addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -671,6 +684,99 @@ var ImageWidget = SearchWidget.extend({
     },
 });
 
+/**
+ * Let users choose an image, including uploading a new image in odoo.
+ */
+var ImageWidget = FileWidget.extend({
+    template: 'wysiwyg.widgets.image',
+
+    /**
+     * @constructor
+     */
+    init: function (parent, media, options) {
+        options = options || {};
+        this._super.apply(this, [parent, media, _.extend({}, options, {
+            accept: options.accept || 'image/*',
+            mimetypeDomain: options.mimetypeDomain || ['mimetype', 'in',
+                ['image/gif', 'image/jpe', 'image/jpeg', 'image/jpg', 'image/gif', 'image/png']
+            ],
+        })]);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _renderExisting: function (rows, withEffect) {
+        return QWeb.render('wysiwyg.widgets.image.existing.attachments', {
```

Comment:
I did it this way in case later we need to pass different parameters to the view depending on each widget.

I'm ok to move it into the abstract, then the only parameter of the method should be the object to pass to the view?

Reply:
Nevermind I understand actually you don't want me to keep the method at all, but then as I said it can't be customized per widget?

Reply:
No, I am ok with keeping the method. What about just adding `widget: this` in the object in all cases so that the view can access the widget attributes if necessary at one point?

Reply:
Ok to add it, I thought it was automatic but I guess that's only true when the view is rendered automatically from the widget template.